instrumentation/instacosmos/container_client.go

```diff
@@ -534,9 +534,11 @@ func (icc *instaContainerClient) setError(s tracing.Span, err error) {
 	})
 
 	// setting status code of error response
-	var responseErr *azcore.ResponseError
-	errors.As(err, &responseErr)
-	icc.setStatus(s, responseErr.StatusCode)
+	var responseErr = new(azcore.ResponseError)
+	ok := errors.As(err, &responseErr)
+	if ok {
```

Comment:
```suggestion
	if ok := errors.As(err, &responseErr); ok { 
```

Suggestion/Nitpicking: You can combine the `if` blocks for simple cases if you wish to